### PR TITLE
update go-libp2p to v0.30.0

### PR DESCRIPTION
**Pull request by marten-seemann**
## Proposed Changes

* Update CI to Go 1.20.7.  At this point, Go 1.19 is unmaintained by the Go team.
* Update go-libp2p to v0.30.0

## Additional Info

[go-libp2p v0.30.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.30.0) allows using Go 1.21. It also drops support for Go 1.19.

Note that this PR does NOT bump the Go version in `go.mod`. I tried that, and it leads to a whole lot of `staticcheck` complaints. I'll leave this as an exercise for the reader :)

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] New features have usage guidelines and / or documentation updates in
  - [x] [Lotus Documentation](https://lotus.filecoin.io)
  - [x] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/store/store.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ import (
 	"sync"
 	"time"
 
-	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/hashicorp/golang-lru/arc/v2"
```
Comment (marten-seemann) [inline chain/store/store.go:14]: golang-lru introduced a breaking change in the v2.0.3 patch release. See https://github.com/hashicorp/golang-lru/issues/142 for discussion. The author has his own interpretation of the semver specification, that diverges from everyone else's reading, and seems oblivious to the amount of pain his actions cause in the ecosystem.

**Comment by marten-seemann** [comment]
Not sure why some of the tests are failing. Flaky? Could anyone more familiar with the code base please take a look?

File: go.mod [hunk 2]
```diff
@@ -295,9 +296,8 @@ require (
 	github.com/prometheus/procfs v0.9.0 // indirect
 	github.com/prometheus/statsd_exporter v0.22.7 // indirect
 	github.com/quic-go/qpack v0.4.0 // indirect
-	github.com/quic-go/qtls-go1-19 v0.3.3 // indirect
-	github.com/quic-go/qtls-go1-20 v0.2.3 // indirect
-	github.com/quic-go/quic-go v0.36.4 // indirect
+	github.com/quic-go/qtls-go1-20 v0.3.2 // indirect
```
[the remark this answers is not in the document]
Comment (marten-seemann) [reply go.mod]: Thanks for catching this! I bumped it to v0.3.3, which removed the test binary. No other changes in that release, so this should be safe to do.

**Review by magik6k — APPROVED** [review]
LGTM, seems to run on mainnet just fine

**Pull request merged**